common/src/main/java/com/wynntils/features/chat/ChatTimestampFeature.java

```diff
@@ -43,20 +47,84 @@ protected void onConfigUpdate(Config<?> config) {
         }
     }
 
-    @SubscribeEvent(priority = EventPriority.LOW)
-    public void onChat(ChatMessageReceivedEvent event) {
-        if (formatter == null) return;
+    // We only format the timestamp when adding the line to improve performance
+    @SubscribeEvent
+    public void onGuiMessageLineAdd(AddGuiMessageLineEvent event) {
+        GuiMessageExtension messageExtension = (GuiMessageExtension) (Object) event.getMessage();
+        ((GuiMessageLineExtension) (Object) event.getLine()).setCreated(messageExtension.getCreated());
 
-        StyledText message = event.getStyledText();
+        if (event.getIndex() != 0 || formatter == null) return;
 
-        LocalDateTime date = LocalDateTime.now();
-        MutableComponent timestamp = Component.empty()
+        GuiMessageLineExtension extension = (GuiMessageLineExtension) (Object) event.getLine();
+
+        extension.setTimestamp(Component.empty()
                 .append(Component.literal("[").withStyle(ChatFormatting.DARK_GRAY))
-                .append(Component.literal(date.format(formatter)).withStyle(ChatFormatting.GRAY))
-                .append(Component.literal("] ").withStyle(ChatFormatting.DARK_GRAY));
+                .append(Component.literal(extension.getCreated().format(formatter))
+                        .withStyle(ChatFormatting.GRAY))
+                .append(Component.literal("] ").withStyle(ChatFormatting.DARK_GRAY)));
+    }
+
+    @SubscribeEvent
+    public void onChatComponentRenderPre(ChatComponentRenderEvent.Pre event) {
+        timestampWidth = 0;
+
+        for (GuiMessage.Line line : ((ChatComponentAccessor) event.getChatComponent()).getTrimmedMessages()) {
+            GuiMessageLineExtension extension = (GuiMessageLineExtension) (Object) line;
+            Optional<Pair<Component, Integer>> timestamp = extension.getTimestamp();
+
+            if (timestamp.isEmpty()) continue;
+
+            timestampWidth = Math.max(timestampWidth, timestamp.get().b());
+        }
+    }
+
+    @SubscribeEvent
+    public void onChatComponentTranslate(ChatComponentRenderEvent.Translate event) {
+        if (timestampWidth != 0) event.setX(event.getX() + 4 + timestampWidth);
+    }
+
+    @SubscribeEvent
+    public void onChatComponentMapMouseX(ChatComponentRenderEvent.MapMouseX event) {
+        if (timestampWidth != 0) event.setX(event.getX() - (4 + timestampWidth));
+    }
```

Comment:
Can we get some 1-liner comments on what these do (and why they differ slightly in calculation)? Also, why you are at it, can you put in braces? We don't like not putting them for anything that's not code flow-control, like return/break/continue.